internal/pipeline/pipeline.go

```diff
@@ -228,6 +247,64 @@ func (f *Function) Validate() error {
 	return nil
 }
 
+// runner returns a fnRunner from the image and configs of
+// this function.
+func (f *Function) runner() (*fnRunner, error) {
+	config, err := f.config()
+	if err != nil {
+		return nil, err
+	}
+	return &fnRunner{
+		fn: &runtime.ContainerFn{
+			Image: f.Image,
+		},
+		fnConfig: config,
+	}, nil
+}
+
+func (f *Function) config() (*yaml.RNode, error) {
+	var dataNode *yaml.RNode
+	switch {
+	case f.ConfigPath != "":
+		file, err := os.Open(f.ConfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open config file path %s: %w", f.ConfigPath, err)
+		}
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read content from config file: %w", err)
+		}
+		dataNode, err = yaml.Parse(string(b))
```

Comment:
I am thinking of a CUJ where someone implements a generator function that takes a KRM file as input (that is CR in this case), wrapping it in a configmap seems odd and will require them to implement extra unwrapping logic in their code.
I think relaying the config content as-is without wrapping is desirable here.